scripts/pre_commit_linter.py

```diff
@@ -2439,9 +2441,67 @@ def _check_constants_declaration(self):
                 filepath for filepath in self.all_filepaths if (
                     filepath.endswith('.ts'))]
             constants_to_source_filepaths_dict = {}
+            angularjs_source_filepaths_to_constants_dict = {}
             for filepath in ts_files_to_check:
-                # Check that the constants are declared only in a *.constants.ts
+                # The following block extracts the corresponding Angularjs
+                # constants file for the Angular constants file. This is
+                # required since the check cannot proceed if the AngularJS
+                # constants file is not provided before the Angular constants
                 # file.
+                if filepath.endswith('.constants.ts'):
+                    filename_without_extension = filepath[:-3]
+                    corresponding_angularjs_filepath = (
+                        filename_without_extension + '.ajs.ts')
+                    compiled_js_dir = tempfile.mkdtemp(dir=os.getcwd())
+                    try:
+                        compiled_js_filepath = self._compile_ts_file(
+                            corresponding_angularjs_filepath, compiled_js_dir)
+                        file_content = FileCache.read(
+                            compiled_js_filepath).decode(
+                                'utf-8')
+
+                        parsed_script = esprima.parseScript(file_content)
+                        parsed_nodes = parsed_script.body
+                        angularjs_constants_set = set()
```

Comment:
Should be a list. Then later you can do len(X) == len(set(X)) to detect duplicates.

Reply:
Done!